src/vstart.sh

```diff
@@ -464,6 +471,40 @@ get_pci_selector() {
     lspci -mm -n -D -d $pci_id | cut -d' ' -f1
 }
 
+do_rgw_conf() {
+
+	if [ $CEPH_NUM_RGW -eq 0 ]; then
+		return 0
+	fi
+
+	if [ $CEPH_NUM_RGW -gt 26 ]; then
+		echo "unable to run more than 26 rgws for vstart.sh's simplicity"
+		CEPH_NUM_RGW=26
+	fi
+
+	# setup each rgw on a sequential port, starting at $CEPH_RGW_PORT. Use an
+	# alphabetical id, sequential starting at 'A', as a suffix to 'host';
+	# i.e., 'hostA', 'hostB', etc.
+	current_port=$CEPH_RGW_PORT
+	for n in $(seq 1 $CEPH_NUM_RGW); do
+		chr=$(printf "$(printf '\\x%02x' $((64+$n)) )")
+		rgw_id=host${chr}
+		wconf << EOF
+[client.rgw.$rgw_id]
+        rgw frontends = $rgw_frontend port=$current_port
+        admin socket = ${CEPH_OUT_DIR}/radosgw.${rgw_id}.asok
+        ; needed for s3tests
+        rgw crypt s3 kms encryption keys = testkey-1=YmluCmJvb3N0CmJvb3N0LWJ1aWxkCmNlcGguY29uZgo= testkey-2=aWIKTWFrZWZpbGUKbWFuCm91dApzcmMKVGVzdGluZwo=
+        rgw crypt require ssl = false
+        ; uncomment the following to set LC days as the value in seconds;
+        ; needed for passing lc time based s3-tests (can be verbose)
+        ; rgw lc debug interval = 10
```

Comment:
the `rgw frontends` and `admin socket` are different between instances, but the rest could stay under `[client.rgw]`

Reply:
I believe the `admin socket` will be different for each instance (given we're specifying dependent on `${rgw_id}`, but yeah, we're assigning the same port to each instance, which will be bad. Would an incremental port be okay, or this have some other behavior?

Reply:
incremental port is how it worked before, and it looks like you're incrementing current_port here too :+1: 

i'm just suggesting that we leave the initial [client.rgw] section for the variables that don't change between instances, and only put `rgw frontends` and `admin socket` in these [client.rgw.$id] sections